src/CellContentParser.ts

```diff
@@ -3,187 +3,216 @@
+       this.value = new CellError(errorType, message)
+     }
+   }
+ 
+   export type Type = Number | String | Boolean | Empty | Formula | Error
+ }
+ 
+ /**
+  * Checks whether string looks like formula or not.
+  *
+  * @param text - formula
+  */
+ export function isFormula(text: string): boolean {
+   return text.startsWith('=')
+ }
+ 
+ export function isBoolean(text: string): boolean {
+   const tl = text.toLowerCase()
+   return tl === 'true' || tl === 'false'
+ }
+ 
+ export function isPercent(text: string): boolean {
+   const trimmedContent = text.trim()
+ 
+   return trimmedContent.endsWith('%')
+ }
+ 
+ export function isError(text: string, errorMapping: Record<string, ErrorType>): boolean {
+   const upperCased = text.toUpperCase()
+   const errorRegex = /#[A-Za-z0-9\/]+[?!]?/
+   return errorRegex.test(upperCased) && Object.prototype.hasOwnProperty.call(errorMapping, upperCased)
+ }
+ 
+ export class CellContentParser {
+   constructor(
+     private readonly config: Config,
+     private readonly dateHelper: DateTimeHelper,
+     private readonly numberLiteralsHelper: NumberLiteralHelper) {
+   }
+ 
+   public parse(content: RawCellContent): CellContent.Type {
+     if (content === undefined || content === null) {
+       return CellContent.Empty.getSingletonInstance()
+     } else if (typeof content === 'number') {
+       if (isNumberOverflow(content)) {
+         return new CellContent.Error(ErrorType.NUM, ErrorMessage.ValueLarge)
+       } else {
+         return new CellContent.Number(content)
+       }
+     } else if (typeof content === 'boolean') {
+       return new CellContent.Boolean(content)
+     } else if (content instanceof Date) {
+       const dateVal = this.dateHelper.dateToNumber({
+         day: content.getDate(),
+         month: content.getMonth() + 1,
+         year: content.getFullYear()
+       })
+       const timeVal = timeToNumber({
+         hours: content.getHours(),
+         minutes: content.getMinutes(),
+         seconds: content.getSeconds() + content.getMilliseconds() / 1000
+       })
+       const val = dateVal + timeVal
+       if (val < 0) {
+         return new CellContent.Error(ErrorType.NUM, ErrorMessage.DateBounds)
+       }
+       if (val % 1 === 0) {
+         return new CellContent.Number(new DateNumber(val, 'Date()'))
+       } else if (val < 1) {
+         return new CellContent.Number(new TimeNumber(val, 'Date()'))
+       } else {
+         return new CellContent.Number(new DateTimeNumber(val, 'Date()'))
+       }
+     } else if (typeof content === 'string') {
+       if (isBoolean(content)) {
+         return new CellContent.Boolean(content.toLowerCase() === 'true')
+       } else if (isFormula(content)) {
+         return new CellContent.Formula(content)
+       } else if (isError(content, this.config.errorMapping)) {
+         return new CellContent.Error(this.config.errorMapping[content.toUpperCase()])
+       } else if (isPercent(content)) {
```

Comment:
I don't know why git is saying this entire file changed. I simply separated out the percent/currency checks into if statements as well instead of a variable condition.